isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc

```diff
@@ -468,6 +468,39 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
 [id="image-CallFAmGuX",reftext="{figure-caption} {counter:figures}"]
 image::blaupausen:referenzarchitektur/CallFAmGuX.png[align="center"]
 
+[[zentrale-sicherheitsaspekte]]
+=== Zentrale Sicherheitsaspekte in der Referenzarchitektur
+
+Sicherheit ist ein Ziel der Referenzarchitektur (<<die-ziele-der-referenzarchitektur>>)
+und technische Querschnittsaufgabe.
+Ein Aspekt davon ist der Schutz vor unberechtigtem Zugriff auf vertrauliche Daten.
+Wie bereits in <<image-GA-AW2>> angedeutet, können Nutzer einer IsyFact-Anwendung nicht direkt darauf zugreifen, sondern nur über eine dazwischenliegenden Informations- und Dienstezone:
+
+- Interne u. externe Anwender mit ihrem Web-Client (Browser) über ein Portal,
+- Externe Anwendungen verbinden sich über Internet-Protokolle mit einem Service-Gateway.
+- In zukünftigen Versionen von IsyFact muss sich eine externe Anwendung erst über das Authentication-Gateway durch den IAM-Service authentifizieren lassen, bevor es über ein Service-Gateway Zugang zu einer IsyFact-Anwendung erhält.
+Aktuell wird diese Funktion vom Service-Gateway mit übernommen.
+
+.Integration von IAM in eine Anwendungslandschaft
+[id="image-iam-integration-awl",reftext="{figure-caption} {counter:figures}"]
+image::referenzarchitektur/sicherheit_iam_integration_awl-2.dn.svg[]
+
+Bevor der Zugang zu der IsyFact-Anwendung gewährt wird, erfolgt in der Informations- und Dienstezone eine Berechtigungsprüfung:
+
+Im Portal bzw. Authentication-Gateway wird die Identität des Nutzers überprüft, indem Authentifizierungsdaten des Nutzers an einen zentralen Service (IAM-Service) in der Anwendungslandschaft weitergereicht werden.
+Ist der Nutzer dem IAM-Service bekannt, stellt er ein Access-Token aus, das auch die dem Nutzer zugeordneten Rollen enthält.
+Die Anfrage des Nutzers wird dann zusammen mit diesem Access-Token an die Interaktions- bzw. Service-Schicht der IsyFact-Fachanwendung weitergeleitet.
+Dort wird (über ein Rolle-Rechte Mapping) geprüft, ob der Nutzer das zur Ausführung des gewünschten Service notwendige Recht besitzt (Autorisierung).
+
+Eine detaillierte Beschreibung der erwähnten Systeme (Portal, Service-Gateway, Authentication-Gateway, IAM) und der Autorisierungs- und Authentifizierungsprotokolle OAuth2.0 und OpenID Connect findet sich in xref:isyfact-standards-doku:isy-security:konzept/master.adoc#inhalt[Konzept des Querschnitt-Bausteins Security].
+
+NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
+
+Weitere Sicherheit bietet die zusätzliche Verwendung eines Client-Zertifikats:
+Die betreibende Behörde stellt ein Client-Zertifikat aus und verteilt es an alle berechtigten Nutzer.
+Ein dem Portal bzw. Authentication-Gateway vorgeschalteter Loadbalancer (nicht im Bild) verlangt und prüft dieses Client-Zertifikat.
```

Comment:
Den Loadbalancer würde hier nicht erwähnen, sondern dass beim Betreten der Anwendungslandschaft das Zertifikat geprüft wird und Informationen aus dem Zertifikat, wie z.B. das Behördenkennzeichen, des Aufrufers zur Authentifizierung oder Autorisierung in der Anwendungslandschaft genutzt werden können. Kurze Erwähnung, dass der Baustein isy-security das aus dem Zertifikat extrahierte BHKNZ als zweiten Faktor in der Authentifizierung verwendet, sofern es verfügbar ist.

Reply:
Eine grobe Autorisierung von Zugriffen sollte ebenfalls an den Außengrenzen geschehen, z.B. durch Client-
Zertifikate.